### PR TITLE
No items spawn near spaceship

### PR DESCRIPTION
**Pull request by AbhiKapoor55**
- We received feedback from Elias last time that there are too many spaceship parts spawning right next to the Spaceship, making it super easy to win the game 
- I changed the spawning algorithm in RandomSpawner.cs so that no spaceship parts are spawned within a radius of 70 around the spaceship

### CODE REVIEW

File: NoMansLand/Assets/Scripts/RandomSpawner.cs [hunk 1]
```diff
@@ -19,9 +19,10 @@ public class RandomSpawner : MonoBehaviour
     public int terrainZLength; // length of terrain in Z axis 
     public int itemXPos; // x position of item 
     public int itemZPos; // z position of item 
-    public const int cushionAmount = 50; // to ensure spawning does not happen at the edges (at the borders)
-
+    public const int cushionAmount = 70; // to ensure spawning does not happen at the edges (at the borders)
     private GameObject sampleObj;
+    public int spaceshipX; 
+    public int spaceshipZ;
```
Comment (slakh96) [inline NoMansLand/Assets/Scripts/RandomSpawner.cs:25]: Do we need these to be public? is anyone else using these values except for this file?

**Review by gazijarin — DISMISSED** [review]
looks good

File: NoMansLand/Assets/Scripts/RandomSpawner.cs [hunk 2]
```diff
@@ -48,10 +52,26 @@ IEnumerator CollectibleDrop()
...
-            itemZPos = UnityEngine.Random.Range(terrainZPos + cushionAmount, terrainZPos + terrainZLength - cushionAmount);
+            // Basically: the "spawnable" region has been split into two areas  
+            // Area 1: edge of map (+ cushion amount) up until spaceship (- cushion amount)
+            // Area 2: spaceship ( + cushion amount) up until other edge of map 
+            // As a result, spaceship +/- cushion amount has been excluded from spawning items 
+            // We will spawn in these two areas equally 
+            if(i%2 == 0) 
+            {
+                // Generate an X position in Area 1 
+                itemXPos = UnityEngine.Random.Range(terrainXPos + cushionAmount, spaceshipX - cushionAmount);
```
Comment (slakh96) [inline NoMansLand/Assets/Scripts/RandomSpawner.cs:63]: Wondering if this algorithm will generate dead spots? Places where parts will not spawn again. reaching out to u on fb

**Review by slakh96 — COMMENTED** [review]
Might be buggy? not sure

File: NoMansLand/Assets/Scripts/RandomSpawner.cs (hunk 2 above)
Comment (slakh96) [reply NoMansLand/Assets/Scripts/RandomSpawner.cs:63]: Update: Resolved on call

**Review by slakh96 — APPROVED** [review]
LGTM

**Pull request merged**